### PR TITLE
CI: Min Pytest Cov Version/Restrict xdist version

### LINKED ISSUE #35737: CI: Travis failing with INTERNALERROR
I'm seeing this on every recent Travis 37-coverage build:

```
============================= test session starts ==============================
platform linux -- Python 3.7.7, pytest-6.0.1, py-1.9.0, pluggy-0.13.1
rootdir: /home/travis/build/pandas-dev/pandas, configfile: setup.cfg, testpaths: pandas
plugins: xdist-2.0.0, cov-2.10.0, forked-1.2.0, hypothesis-5.24.0
gw0 I
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/_pytest/main.py", line 238, in wrap_session
INTERNALERROR>     config.hook.pytest_sessionstart(session=session)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/manager.py", line 87, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/xdist/dsession.py", line 78, in pytest_sessionstart
INTERNALERROR>     nodes = self.nodemanager.setup_nodes(putevent=self.queue.put)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/xdist/workermanage.py", line 65, in setup_nodes
INTERNALERROR>     return [self.setup_node(spec, putevent) for spec in self.specs]
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/xdist/workermanage.py", line 65, in <listcomp>
INTERNALERROR>     return [self.setup_node(spec, putevent) for spec in self.specs]
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/xdist/workermanage.py", line 73, in setup_node
INTERNALERROR>     node.setup()
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/xdist/workermanage.py", line 260, in setup
INTERNALERROR>     self.config.hook.pytest_configure_node(node=self)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/manager.py", line 87, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pytest_cov/plugin.py", line 239, in pytest_configure_node
INTERNALERROR>     self.cov_controller.configure_node(node)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pytest_cov/engine.py", line 274, in configure_node
INTERNALERROR>     workerinput(node).update({
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pytest_cov/compat.py", line 42, in fn
INTERNALERROR>     return getattr(obj, attr, *args)
INTERNALERROR> AttributeError: 'WorkerController' object has no attribute 'slaveinput'
The command "ci/run_tests.sh" exited with 3.

```

### PR DESCRIPTION
**Pull request by alimcmaster1**
- [x] closes #35737

### CODE REVIEW

**Comment by rhshadrach** [comment]
Travis was successful, although I'm not sure why 2.10.1 isn't pulled in without this change. Any idea there?

**Comment by alimcmaster1** [comment]
cc @jreback issue created for unpin: https://github.com/pandas-dev/pandas/issues/35756

File: ci/deps/azure-windows-38.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ dependencies:
   # tools
   - cython>=0.29.16
   - pytest>=5.0.1
-  - pytest-xdist>=1.21
+  - pytest-xdist>=1.21,<2.0.0 #GH 35737
```
Comment (jbrockmendel) [inline ci/deps/azure-windows-38.yaml:11]: space after `#` pls

**Comment by simonjayhawkins** [comment]
Thanks @alimcmaster1

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Pull request merged**